build/build.sh

```diff
@@ -3,12 +3,29 @@
 set -e
 
 OSDIST=`grep '^ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"'`
+OSVERSION=`grep '^VERSION_ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"'`
+OSMAJOR=${OSVERSION%.*}
+GCCVERSION=`gcc -v 2>&1 >/dev/null | grep 'gcc version' | awk '{print $3}'`
+GCCMAJOR=${GCCVERSION:0:1}
 BUILDDIR=$(readlink -f $(dirname ${BASH_SOURCE[0]}))
 CORE=`grep -c ^processor /proc/cpuinfo`
 CMAKE=cmake
 CMAKE_MAJOR_VERSION=`cmake --version | head -n 1 | awk '{print $3}' |awk -F. '{print $1}'`
 CPU=`uname -m`
 
+if [[ ( $OSDIST == "rhel" || $OSDIST == "centos" ) && $OSMAJOR -lt "8" && $GCCMAJOR -lt "9" ]]; then
+    # RHEL/CentOS 7.x does not have C++17 support by default
+    # Attempt fallback option with devtoolset-9
+    echo "Trying devtoolset-9"
+    {
+        # Try
+	source scl_source enable devtoolset-9
+    } || {
+        # Except
+	echo "Could not find a suitable gcc version" && exit 1
+    }
+fi
+
```

Comment:
And what if I am compiling with clang?

@bryanloz-xilinx,  I don't think we should do this.  If you can get CMake to error out when the picked up C++ compiler doesn't support c++17, then that would be great.  But I don't think changing the build script is not a good idea.    Developers / and folks who build XRT should just follow the instructions that you nicely changed.   You can add your own checks in your own set up scripts.

Reply:
Sure. Thanks!
We do have at least one external user building with clang but undoubtedly on an obsolete OS.
https://github.com/Xilinx/XRT/pull/6391

Reply:
CXX 17 is already a strict requirement and the error message is quite specific about this.

Reply:
> CXX 17 is already a strict requirement and the error message is quite specific about this.

For some reason, I thought the errors were more catastrophic looking in the past, but I agree, they look okay in the modern day:

```
CMake Error in runtime_src/core/pcie/linux/CMakeLists.txt:
  Target "xrt_core" requires the language dialect "CXX17" (with compiler
  extensions), but CMake does not know the compile flags to use to enable it.
```